libcodechecker/log/option_parser.py

```diff
@@ -308,218 +200,228 @@ def item(self):
+    # command with different languages.
+    if flag_iterator.item == '-x':
+        target.append('-x')
+        next(flag_iterator)
+        target.append(flag_iterator.item)
+        buildaction.lang = flag_iterator.item
+        return True
+
+    return False
+
+
+def __get_output(flag_iterator, target, buildaction):
+    """
+    This function consumes -o flag which is followed by the output file of the
+    action. This file is then collected to the buildaction object and saved to
+    the target command.
+    """
+    if flag_iterator.item == '-o':
+        target.append('-o')
+        next(flag_iterator)
+        target.append(flag_iterator.item)
+        buildaction.output = flag_iterator.item
+        return True
+
+    return False
+
+
+def __replace(flag_iterator, target, buildaction):
+    """
+    This function extends the target list with the corresponding replacement
+    based on REPLACE_OPTIONS_MAP if the flag_iterator is currently pointing to
+    a flag to replace.
+    """
+    value = REPLACE_OPTIONS_MAP.get(flag_iterator.item)
+
+    if value:
+        target.extend(value)
+
+    return bool(value)
+
+
+def __skip(flag_iterator, target, buildaction):
+    """
+    This function skips the flag pointed by the given flag_iterator with its
+    parameters if any.
+    """
+    if IGNORED_OPTIONS.match(flag_iterator.item):
+        return True
+
+    for pattern, arg_num in IGNORED_PARAM_OPTIONS.items():
+        if re.match(pattern, flag_iterator.item):
+            for i in range(arg_num):
+                next(flag_iterator)
+            return True
+
+    return False
+
+
+def parse_options(gcc_command):
+    """
+    This function parses a GCC compilation action and returns a BuildAction
+    object which can be the input of Clang analyzer tools.
+
+    gcc_command -- A valid GCC build command.
+    """
+
+    buildaction = BuildAction()
+
+    # Keep " characters.
+    gcc_command = gcc_command.replace(r'"', r'"\"')
+    gcc_command = shlex.split(gcc_command)
+
+    buildaction.action_type = buildaction.COMPILE
+    buildaction.compiler = gcc_command[0]
+    if '++' in buildaction.compiler or 'cpp' in buildaction.compiler:
+        buildaction.lang = 'c++'
+
+    clang_command = [gcc_command[0]]
+
+    flag_transformers = [
+        __get_arch,
+        __get_language,
+        __get_output,
+        __determine_action_type,
+        __replace,
+        __skip,
+        __append_from_file,
+        __collect_compile_opts]
+
+    for it in OptionIterator(gcc_command[1:]):
+        for flag_transformer in flag_transformers:
+            if flag_transformer(it, clang_command, buildaction):
+                break
+        else:
+            clang_command.append(it.item)
+            #print('Unhandled argument: ' + it.item)
+
+    # TODO: It is difficult to fetch source files from a compilation command if
+    # they contain a space character.
+    #for source_file in buildaction.sources:
```

Comment:
Is this a regression? Or this was buggy before? I think if source files contains spaces they should be either escaped or in quotes. Shouldn't this already be handled by shsplit or some similar facility?

Reply:
This is an outdated modification too. But the issue is relevant, this will go to a new ticket about handling spaces in file names by the logger.
(Side note: what about reverse compatibility with older log files? This extra logic seems to be necessary for the first view, but we'll see later.)

Reply:
I think we should not care about reverse compatibility in the case. In CI loops the users should always regenerate the build log since files might have been removed or added in commits. I cannot really see a usecase where the users are not generating new compilation database unless someone is running a buildbot for codechecker itself.

Reply:
I don't know... In `tests/unit/logparser_test_files` you can see what kind of old log file formats are supported. I think the idea of CodeChecker is that it is an infrastructure of multiple tools. It supports several analyzers and it seems that is supports several logger tools as well, not just the delivered one.